jstz_cli/src/run.rs

```diff
@@ -18,12 +20,49 @@ pub async fn exec(
     http_method: String,
     json_data: Option<String>,
 ) -> Result<()> {
+    // Resolve URL
+    let mut url_object =
+        Url::parse(&url).map_err(|e| anyhow!("Failed to parse URL: {}", e))?;
+    if let Some(host) = url_object.host_str() {
+        if !host.starts_with("tz4") {
+            if cfg.accounts().contains(&host) {
+                url_object
+                    .set_host(Some(
+                        cfg.accounts().get(host)?.address().to_base58().as_str(),
+                    ))
+                    .map_err(|_| anyhow!("Failed to set host"))?;
+            } else {
+                return Err(anyhow!("No such account exists."));
+            }
+        }
+    } else {
+        return Err(anyhow!("Invalid URL"));
+    }
+
+    let resolved_url = url_object.to_string();
+    println!("Resolved URL: {}", resolved_url);
+
     let account = cfg.accounts.account_or_current_mut(referrer)?;
+    let (nonce,
+        _alias,
+        address,
+        secret_key,
+        public_key
+    ) = match account {
+        Account::Owned(OwnedAccount{
+            nonce,
+            alias,
+            address,
+            secret_key,
+            public_key,
+         }) => (nonce, alias.clone(), address.clone(), secret_key.clone(), public_key.clone()),
+        _ => return Err(anyhow!("The account is an alias and cannot be used for the run of a smart function. Please use an owned account.")),
+    };
```

Comment:
```suggestion
     let OwnedAccount { nonce, address, secret_key, public_key, alias: _ } =
      cfg.accounts().get(alias)?.into_owned_account()?;
```

Reply:
Or 
```suggestion
    let account = cfg.accounts().get(alias)?.into_owned_account()?;
```
And use `account` as we previously did